File: app/src/main/java/com/tt/handsomeman/di/module/StartScreenModule.java
package com.tt.handsomeman.di.module;

import com.tt.handsomeman.service.StartScreenService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public class StartScreenModule {

    @Provides
    @Singleton
    StartScreenService provideStartScreenService(Retrofit retrofit) {
        return retrofit.create(StartScreenService.class);
    }
}

File: app/src/main/java/com/tt/handsomeman/di/module/StartScreenModule.kt
package com.tt.handsomeman.di.module

import com.tt.handsomeman.service.StartScreenService

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

@Module
class StartScreenModule {

    @Provides
    @Singleton
    internal fun provideStartScreenService(retrofit: Retrofit): StartScreenService {
        return retrofit.create(StartScreenService::class.java!!)
    }
}
